src/MultiTypeSymbolTable.chpl

```diff
@@ -146,6 +146,14 @@ module MultiTypeSymbolTable
                 if (v) {writeln("deleteEntry: unkown symbol ",name);try! stdout.flush();}
             }
         }
+
+        /*
+        Clears all unregistered entries from the symTable
+        */
+        proc clear() {
+            for n in tab { deleteEntry(n); }
```

Comment:
I don't think this idiom is legal since you're removing from a map (which could shrink it) while iterating over it.

I think `for n in tab.keysToArray() { deleteEntry(n); }` would work though. 

@daviditen and/or @dlongnecke-cray would know best though.

Reply:
@bradcray @daviditen @ronawho is this a bug? really?

Reply:
Most languages don't allow this. e.g. you can't remove from a python dictionary while iterating over it:

```python
mydict = {'one': 1, 'two': 2, 'three': 3}
for k in mydict.keys():
   if k == 'two':
       del mydict[k]
```

```
RuntimeError: dictionary changed size during iteration
```

But you can do `for k in list(mydict.keys()):`

Reply:
I agree that removing from a map while iterating over it is problematic. It's iterating over a fixed range while potentially shrinking the storage the range represents.

Reply:
ok, i'll make the change.